crates/api/src/content.rs

```diff
@@ -15,7 +15,7 @@ pub struct ContentSource {
 
 /// Represents the supported kinds of content sources.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(tag = "type", rename_all = "camelCase")]
+#[serde(tag = "type", rename_all = "camelCase", content = "c")]
```

Comment:
Rather than use this type of tagged enum, I think we should just go back to having `HttpAnonymous` be a struct -like variant with a `url` field.

Sorry about breaking this when I did the api crate refactoring.

Reply:
I missed this change in your refactor. Agree that we should bring back the struct-like variant and url field. Other than that this PR lgtm.

Reply:
Ah no worries about the change.  Pushed a commit to bring it back.